kolibri/plugins/management/assets/src/actions.js

```diff
@@ -25,19 +25,30 @@ const samePageCheckGenerator = require('kolibri.coreVue.vuex.actions').samePageC
  * the API to state in the Vuex store
  */
 
-function _userState(data) {
-  // assume just one role for now
+function _stateUser(apiUserData) {
+  // handle role representation
   let kind = UserKinds.LEARNER;
-  if (data.roles.length && data.roles[0].kind === 'admin') {
-    kind = UserKinds.ADMIN;
-  }
+
+  // makes kind the highest ranking role
+  apiUserData.roles.forEach(role => {
+    // using a switch statement. Checks all in order of heirarchy
+    switch(role.kind){
+      // sets role to admin 
+      case UserKinds.ADMIN || UserKinds.SUPERUSER:
+        kind = UserKinds.ADMIN;
+        break;
+      case UserKinds.COACH:
+        if(kind != UserKinds.ADMIN) kind = UserKinds.COACH;
+        break;
+    }
+  });
```

Comment:
okay, the suggestion given might not work well then. regarding the existing code:

```
  let kind = UserKinds.LEARNER;

  // makes kind the highest ranking role
  apiUserData.roles.forEach(role => {
    // using a switch statement. Checks all in order of heirarchy
    switch(role.kind){
      // sets role to admin 
      case UserKinds.ADMIN || UserKinds.SUPERUSER:
        kind = UserKinds.ADMIN;
        break;
      case UserKinds.COACH:
        if(kind != UserKinds.ADMIN) kind = UserKinds.COACH;
        break;
    }
  });
```

It's difficult to reason about what the resulting `kind` is going to be, because you have to mentally loop through and keep track of prior state. The fact that we have to check whether `kind != UserKinds.ADMIN` inside the second switch case it the primary code smell here. In other words, it would be better if the logic halted as soon as the correct match was found.

One other thing that's adding complexity here is that we're mapping both `ADMIN` and `SUPERUSER` to `ADMIN`. I was wondering why we don't retain the `SUPERUSER` information?